central/sensor/service/pipeline/complianceoperatorresults/pipeline.go

```diff
@@ -153,27 +148,24 @@ func (s *pipelineImpl) processV2ComplianceResult(ctx context.Context, event *cen
 
 	switch event.GetAction() {
 	case central.ResourceAction_REMOVE_RESOURCE:
+		// V1 still needs to function so remove it too
+		if err := s.datastore.Delete(ctx, event.GetId()); err != nil {
+			return err
+		}
+
 		// use V2 datastore
 		return s.v2Datastore.DeleteResult(ctx, event.GetId())
 	default:
-		convertedResult, err := s.convertSensorMsgToV2Storage(ctx, checkResult, clusterID)
-		if err != nil {
+		// Still need to store the V1 version to maintain both
+		if err := s.datastore.Upsert(ctx, internaltov1storage.ConvertInternalToV1Storage(checkResult)); err != nil {
 			return err
 		}
-		return s.v2Datastore.UpsertResult(ctx, convertedResult)
-	}
-}
 
-func (s *pipelineImpl) convertSensorMsgToV2Storage(ctx context.Context, sensorData *central.ComplianceOperatorCheckResultV2, clusterID string) (*storage.ComplianceOperatorCheckResultV2, error) {
-	scanConfigs, err := s.scanConfigDatastore.GetScanConfigurations(ctx, search.NewQueryBuilder().
-		AddExactMatches(search.ComplianceOperatorScanName, sensorData.GetSuiteName()).ProtoQuery())
-	if err != nil {
-		return nil, err
-	}
-	if len(scanConfigs) != 1 {
-		return nil, errors.Errorf("Unable to find matching scan configuration for scan %q", sensorData.GetSuiteName())
+		return s.v2Datastore.UpsertResult(ctx, s.convertSensorMsgToV2Storage(checkResult, clusterID))
```

Comment:
[nit] The naming here confused me at the begging. Maybe it would be good to renaming the functions since both receive the same parameter. 

For example:
```golang
if err := s.datastore.Upsert(ctx, internaltov1storage.ConvertInternalV2ToV1Storage(checkResult)); err != nil {
	return err
}

return s.v2Datastore.UpsertResult(ctx, s.ConvertInternalV2ToV2Storage(checkResult, clusterID))
```